### PR TITLE
#419 Write the original scenario filename to the spawnmap.ini [Basic] section.

### PR DESCRIPTION
**Pull request by devo1929**
This is based on PR https://github.com/CnCNet/xna-cncnet-client/pull/420
The second commit here is the only change. @CCHyper

### CODE REVIEW

**Review by CCHyper — APPROVED** [review]

File: DXMainClient/DXGUI/Multiplayer/GameLobby/GameLobbyBase.cs [hunk 1]
```diff
@@ -1520,11 +1521,96 @@ private void WriteMap(PlayerHouseInfo[] houseInfos)
...
+                    string errorMessage = "Unable to copy supplemental map file".L10N("Client:Main:SupplementalFileCopyError") + $" {file}";
+                    Logger.Log(errorMessage);
+                    Logger.Log(e.Message);
+                    XNAMessageBox.Show(WindowManager, "Error".L10N("Client:Main:Error"), errorMessage);
+                    
+                }
+            }
+            
+            // Write the supplemental map files to the INI (eventual spawnmap.ini)
+            mapIni.SetStringValue("Basic", "SupplementalFiles", string.Join(",", supplementalFileNames));
```
Comment (Rans4ckeR) [inline DXMainClient/DXGUI/Multiplayer/GameLobby/GameLobbyBase.cs:1571]:
```suggestion
            mapIni.SetStringValue("Basic", "SupplementalFiles", string.Join(',', supplementalFileNames));
```

File: DXMainClient/DXGUI/Multiplayer/GameLobby/GameLobbyBase.cs [hunk 2]
```diff
@@ -1520,11 +1521,96 @@ private void WriteMap(PlayerHouseInfo[] houseInfos)
...
+                    XNAMessageBox.Show(WindowManager, "Error".L10N("Client:Main:Error"), errorMessage);
+                }
+            }
+        }
+
+        private static IEnumerable<string> GetSupplementalMapFiles(string basePath, string baseFileName)
+        {
+            // Get the supplemental file names for allowable extensions
+            var supplementalMapFileNames = ClientConfiguration.Instance.SupplementalMapFileExtensions
+                .Select(ext => $"{baseFileName}.{ext}".ToLower())
```
Comment (Rans4ckeR) [inline DXMainClient/DXGUI/Multiplayer/GameLobby/GameLobbyBase.cs:1603]:
```suggestion
                .Select(ext => $"{baseFileName}.{ext}".ToUpperInvariant())
```

File: DXMainClient/DXGUI/Multiplayer/GameLobby/GameLobbyBase.cs [hunk 3]
```diff
@@ -1520,11 +1521,96 @@ private void WriteMap(PlayerHouseInfo[] houseInfos)
...
+                .ToList();
+
+            if (!supplementalMapFileNames.Any())
+                return new List<string>();
+
+            // Get full file paths for all possible supplemental files
+            return Directory.GetFiles(basePath, $"{baseFileName}.*")
+                .Where(f => supplementalMapFileNames.Contains(Path.GetFileName(f).ToLower()));
```
Comment (Rans4ckeR) [inline DXMainClient/DXGUI/Multiplayer/GameLobby/GameLobbyBase.cs:1611]:
```suggestion
                .Where(f => supplementalMapFileNames.Contains(Path.GetFileName(f).ToUpperInvariant()));
```

File: DXMainClient/DXGUI/Multiplayer/GameLobby/GameLobbyBase.cs (hunk 2 above)
Comment (Rans4ckeR) [reply DXMainClient/DXGUI/Multiplayer/GameLobby/GameLobbyBase.cs:1603]: Check out [Globalization rules](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/globalization-warnings) for more info.

**Review by Rans4ckeR — CHANGES_REQUESTED** [review]

**Pull request merged**